src/ocean/task/TaskPool.d

```diff
@@ -361,3 +371,44 @@ unittest
     theScheduler.schedule(new MainTask);
     theScheduler.eventLoop();
 }
+
+unittest
+{
+    // Test that 'awaitRunningTasks()' cannot be called from a task that itself
+    // belongs to the pool.
+
+    static class AwaitTask : Task
+    {
+        void delegate () non_throwing_dg;
+        void delegate () throwing_dg;
+
+        public void copyArguments ( void delegate () non_throwing_dg,
+            void delegate () throwing_dg )
+        {
+            this.non_throwing_dg = non_throwing_dg;
+            this.throwing_dg = throwing_dg;
+        }
+
+        override public void run ( )
+        {
+            this.non_throwing_dg();
```

Comment:
Also can copy pool reference instead of delegate and call `pool.awaitRunningTasks` directly from `run`